server/src/main/java/org/opensearch/index/mapper/StringFieldType.java

```diff
@@ -102,6 +103,35 @@ public Query fuzzyQuery(
         );
     }
 
+    @Override
+    public Query fuzzyQuery(
+        Object value,
+        Fuzziness fuzziness,
+        int prefixLength,
+        int maxExpansions,
+        boolean transpositions,
+        MultiTermQuery.RewriteMethod method,
+        QueryShardContext context
+    ) {
+        if (!context.allowExpensiveQueries()) {
+            throw new OpenSearchException(
```

Comment:
This path still needs a test, or is there one I am missing?

Reply:
@dblock we do have tests already that catch this 

https://github.com/opensearch-project/OpenSearch/blob/3a0c0c0b38c0b42bc519c3673d5cd4a1e3379550/server/src/test/java/org/opensearch/index/mapper/KeywordFieldTypeTests.java#L291C1-L295C132